straxen/plugins/position_reconstruction.py

```diff
@@ -203,3 +204,95 @@ def compute(self, peaks):
         for xy in ('x', 'y'):
             result[xy] = peaks[f'{xy}_{algorithm}']
         return result
+
+    
+    
+@export
+@strax.takes_config(
+    strax.Option('recon_alg_included', help='The list of all reconstruction algorithm considered.',
+                 default=('_mlp','_gcn','_cnn'),
+                )
+)
+class S2ReconPosDiff(strax.Plugin):
+    '''
+    Plugin that provides position reconstruction difference for S2s in events, see note: 
+    https://xe1t-wiki.lngs.infn.it/doku.php?id=xenon:shengchao:sr0:reconstruction_quality
+    '''
+    
+    __version__ = '0.0.2'
+    parallel = True
+    depends_on = 'event_basics'
+    provides = 's2_recon_pos_diff'
+    
+    def infer_dtype(self):
+        dtype =[
+        ('s2_recon_avg_x' ,np.float32,
+         'Mean value of x for main S2'),
+        ('alt_s2_recon_avg_x' ,np.float32,
+         'Mean value of x for alternatice S2'),
+        ('s2_recon_avg_y' ,np.float32,
+         'Mean value of y for main S2'),
+        ('alt_s2_recon_avg_y' ,np.float32,
+         'Mean value of y for alternatice S2'),
+        ('s2_recon_pos_diff' ,np.float32,
+         'Reconstructed position difference for main S2'),
+        ('alt_s2_recon_pos_diff' ,np.float32,
+         'Reconstructed position difference for alternative S2'),
+
+    ]
+        dtype += strax.time_fields
+        return dtype
+    
+    def compute(self,events):
+        
+        result = np.zeros(len(events), dtype=self.dtype)
+        result['time'] = events['time']
+        result['endtime'] = strax.endtime(events)
+        # Computing position difference
+        self.compute_pos_diff(events, result)
+        return result  
+
+    def cal_avg_and_std(self, values, axis=1):
+        average = np.average(values,axis=axis)

```

Comment:
Average is usually associated with a weighted mean. Since you are not using any weights here I would use for clarity `np.mean` instead.